analyzer/src/main/kotlin/managers/utils/NodeSupport.kt

```diff
@@ -213,3 +245,46 @@ private fun getWorkspaceMatchers(definitionFile: File): List<PathMatcher> {
         FileSystems.getDefault().getPathMatcher(pattern)
     }.orEmpty()
 }
+
+private fun getPnpmWorkspaceMatchers(definitionFile: File): List<PathMatcher> {
+    data class PnpmWorkspaces(val packages: List<String>)
+
+    val pnpmWorkspaceFile = definitionFile.parentFile.resolve("pnpm-workspace.yaml")
+
+    return if (pnpmWorkspaceFile.isFile) {
+        val workspaceMatchers = pnpmWorkspaceFile.readValue<PnpmWorkspaces>().packages
+
+        workspaceMatchers.map { matcher ->
+            val pattern = "glob:${definitionFile.parentFile.invariantSeparatorsPath}/${matcher.removeSuffix("/")}"
+            FileSystems.getDefault().getPathMatcher(pattern)
+        }
+    } else {
+        emptyList()
+    }
+}
+
+private fun getPnpmWorkspaceSubmodules(definitionFiles: Set<File>): Set<File> {
+    val result = mutableSetOf<File>()
+
+    definitionFiles.forEach { definitionFile ->
+        val pnpmWorkspacesFile = definitionFile.parentFile.resolve("pnpm-workspace.yaml")
+
+        if (pnpmWorkspacesFile.isFile) {
+            val pathMatchers = getPnpmWorkspaceMatchers(definitionFile)
+
+            pathMatchers.forEach { matcher ->
+                definitionFiles.forEach inner@{ other ->
```

Comment:
Nit: I find these `return` statements to labels very hard to read - they are a bit like the `goto` style in BASIC. Is there an alternative to express this logic?